File: csharp/Resources/VanityDomainsResource.cs
using System.Net.Http;
using System.Threading.Tasks;
using Newtonsoft.Json.Linq;

namespace Authdog
{
    /// <summary>
    /// Environment vanity-domain Wave 3 operations
    /// </summary>
    public class VanityDomainsResource
    {
        private readonly AuthdogClient _client;

        public VanityDomainsResource(AuthdogClient client)
        {
            _client = client;
        }

        public Task<JObject> ListAsync(string tenantId, string environmentId) =>
            _client.RequestAsync<JObject>(
                HttpMethod.Get,
                $"{AuthdogClient.Env(tenantId, environmentId)}/vanity-domains");

        public JObject List(string tenantId, string environmentId) =>
            ListAsync(tenantId, environmentId).GetAwaiter().GetResult();

        public Task<JObject> CreateAsync(string tenantId, string environmentId, object body) =>
            _client.RequestAsync<JObject>(
                HttpMethod.Post,
                $"{AuthdogClient.Env(tenantId, environmentId)}/vanity-domains",
                body);

        public JObject Create(string tenantId, string environmentId, object body) =>
            CreateAsync(tenantId, environmentId, body).GetAwaiter().GetResult();

        public Task<JObject> DeleteAsync(string tenantId, string environmentId, string domainId) =>
            _client.RequestAsync<JObject>(
                HttpMethod.Delete,
                $"{AuthdogClient.Env(tenantId, environmentId)}/vanity-domains/{domainId}");

        public JObject Delete(string tenantId, string environmentId, string domainId) =>
            DeleteAsync(tenantId, environmentId, domainId).GetAwaiter().GetResult();

        public Task<JObject> CheckAsync(string tenantId, string environmentId, string domainId) =>
            _client.RequestAsync<JObject>(
                HttpMethod.Post,
                $"{AuthdogClient.Env(tenantId, environmentId)}/vanity-domains/{domainId}/check");

        public JObject Check(string tenantId, string environmentId, string domainId) =>
            CheckAsync(tenantId, environmentId, domainId).GetAwaiter().GetResult();
    }
}

File: java/src/main/java/com/authdog/resources/VanityDomainsResource.java
package com.authdog.resources;

import com.authdog.AuthdogClient;
import com.authdog.exceptions.ApiException;
import com.authdog.exceptions.AuthenticationException;
import com.fasterxml.jackson.databind.JsonNode;

/**
 * Environment vanity domain operations.
 */
public final class VanityDomainsResource {
    /**
     * Authdog client.
     */
    private final AuthdogClient client;

    /**
     * Create a vanity domains helper.
     * @param clientParam Authdog client
     */
    public VanityDomainsResource(final AuthdogClient clientParam) {
        this.client = clientParam;
    }

    /**
     * List vanity domains.
     * @param tenantId tenant ID
     * @param environmentId environment ID
     * @return domains envelope
     * @throws AuthenticationException when unauthorized
     * @throws ApiException when the request fails
     */
    public JsonNode list(final String tenantId,
                         final String environmentId)
            throws AuthenticationException, ApiException {
        return client.request("GET",
                EnvPaths.prefix(tenantId, environmentId)
                        + "/vanity-domains",
                null, null, JsonNode.class);
    }

    /**
     * Create a vanity domain.
     * @param tenantId tenant ID
     * @param environmentId environment ID
     * @param body request body
     * @return created domain envelope
     * @throws AuthenticationException when unauthorized
     * @throws ApiException when the request fails
     */
    public JsonNode create(final String tenantId,
                           final String environmentId,
                           final Object body)
            throws AuthenticationException, ApiException {
        return client.request("POST",
                EnvPaths.prefix(tenantId, environmentId)
                        + "/vanity-domains",
                body, null, JsonNode.class);
    }

    /**
     * Delete a vanity domain.
     * @param tenantId tenant ID
     * @param environmentId environment ID
     * @param domainId domain ID
     * @return delete envelope
     * @throws AuthenticationException when unauthorized
     * @throws ApiException when the request fails
     */
    public JsonNode delete(final String tenantId,
                           final String environmentId,
                           final String domainId)
            throws AuthenticationException, ApiException {
        return client.request("DELETE",
                EnvPaths.prefix(tenantId, environmentId)
                        + "/vanity-domains/" + domainId,
                null, null, JsonNode.class);
    }

    /**
     * Check vanity domain DNS status.
     * @param tenantId tenant ID
     * @param environmentId environment ID
     * @param domainId domain ID
     * @return check envelope
     * @throws AuthenticationException when unauthorized
     * @throws ApiException when the request fails
     */
    public JsonNode check(final String tenantId,
                          final String environmentId,
                          final String domainId)
            throws AuthenticationException, ApiException {
        return client.request("POST",
                EnvPaths.prefix(tenantId, environmentId)
                        + "/vanity-domains/" + domainId + "/check",
                null, null, JsonNode.class);
    }
}
